e2e-tests/gef/cypress/e2e/facility/facility-end-date.spec.js

```diff
@@ -0,0 +1,250 @@
+      cy.visit(relative(`/gef/application-details/${application.id}/facilities/${facilityId}/about-facility`));
+
+      aboutFacility.isUsingFacilityEndDateYes().click();
+      aboutFacility.saveAndReturnButton().click();
+
+      cy.visit(relative(`/gef/application-details/${application.id}/facilities/${facilityId}/facility-end-date`));
+
+      facilityEndDate.backLink();
+      facilityEndDate.headingCaption();
+      facilityEndDate.facilityEndDateDay();
+      facilityEndDate.facilityEndDateMonth();
+      facilityEndDate.facilityEndDateYear();
+      facilityEndDate.facilityEndDateDetails();
+      facilityEndDate.continueButton();
+      facilityEndDate.saveAndReturnButton();
+    });
+
+    it('redirects the user to `About this facility` page when clicking on back link', () => {
+      cy.visit(relative(`/gef/application-details/${application.id}/facilities/${facilityId}/facility-end-date`));
+
+      facilityEndDate.backLink().click();
+
+      cy.url().should('eq', relative(`/gef/application-details/${application.id}/facilities/${facilityId}/about-facility`));
+    });
+
+    it('validates form when clicking on Continue', () => {
+      cy.visit(relative(`/gef/application-details/${application.id}/facilities/${facilityId}/facility-end-date`));
+
+      facilityEndDate.continueButton().click();
+      facilityEndDate.errorSummary();
+      facilityEndDate.facilityEndDateError();
+    });
+
+    it('redirects user to application page when clicking on `save and return` button', () => {
+      cy.visit(relative(`/gef/application-details/${application.id}/facilities/${facilityId}/facility-end-date`));
+      facilityEndDate.saveAndReturnButton().click();
+      cy.url().should('eq', relative(`/gef/application-details/${application.id}`));
+    });
+
+    it('validates form if not blank when clicking on `save and return` button', () => {
+      cy.visit(relative(`/gef/application-details/${application.id}/facilities/${facilityId}/facility-end-date`));
+
+      facilityEndDate.facilityEndDateDay().clear().type(todayDay);
+      facilityEndDate.facilityEndDateMonth().clear();
+      facilityEndDate.saveAndReturnButton().click();
+      facilityEndDate.errorSummary();
+      facilityEndDate.facilityEndDateError();
+    });
+
+    it('redirects user to application page when clicking on `save and return` button and form has been successfully filled in', () => {
+      cy.visit(relative(`/gef/application-details/${application.id}/facilities/${facilityId}/facility-end-date`));
+
+      facilityEndDate.facilityEndDateDay().clear().type(todayDay);
+      facilityEndDate.facilityEndDateMonth().clear().type(todayMonth);
+      facilityEndDate.facilityEndDateYear().clear().type(nextYear);
+
+      facilityEndDate.saveAndReturnButton().click();
+
+      cy.url().should('eq', relative(`/gef/application-details/${application.id}`));
+    });
+
+    it('when cover start date is given, it validates facility end date is after the cover start date', () => {
```

Comment:
odd q if i set `cover start date` and then set `facility end date` and then go change `cover start date` to after `facility end date` what happens

Reply:
I asked Ahila about this, she said she'd get back to me with a new story for this journey. I'll chase up on this